activerecord/test/cases/adapter_test.rb

```diff
@@ -211,6 +211,28 @@ def test_numeric_value_out_of_ranges_are_translated_to_specific_exception
       end
     end
 
+    def test_exceptions_from_notifications_are_not_translated
+      original_error = StandardError.new("This StandardError shouldn't get translated")
+      subscriber = ActiveSupport::Notifications.subscribe("sql.active_record") { raise original_error }
+      actual_error = assert_raises(StandardError) do
+        @connection.execute("SELECT * FROM posts")
+      end
+
+      assert_equal original_error, actual_error
+
+    ensure
+      ActiveSupport::Notifications.unsubscribe(subscriber) if subscriber
+    end
+
+    def test_other_exceptions_are_translated_to_statement_invalid
```

Comment:
"Other exceptions" is a bit vague - we should specify that this test covers errors raised by the adapter when attempting to execute the query.